### PR TITLE
fixed:#1887 Event and Holiday Calendar View 

### LINKED ISSUE #1887: ADMIN & USER REDESIGN - List of events and holidays in calender view of events.
**Is your feature request related to a problem? Please describe.**
Currently, the event calendar does not provides the list of events and holidays.

**Describe the solution you'd like**

1. Adding a list of holidays, and
2. adding list events below the event calendar.
3. Using color coding for events

**Describe alternatives you've considered**
NA

**Additional context**

**Current**
Talawa-User
Screenshot from 2024-04-07 06-03-11
Talawa-Admin
Screenshot from 2024-04-07 05-45-06

We already figma design made for this.

Screenshot from 2024-04-07 05-46-34

Figma link: https://www.figma.com/file/dTGr5CnGHvkilU5nwnRXg2/Talawa-Admin-UI-Redesign?type=design&node-id=910-355&mode=design&t=EVDV5SZ2cKnOAc2R-0

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by gurramkarthiknetha**
This feature displays an interactive calendar view for tracking events and holidays. The calendar highlights holidays and user/organization-created events for the selected month, with color-coded markers for easy identification

**What kind of change does this PR introduce?**
This PR introduces a new feature for displaying an interactive event calendar with a list of holidays and user/organization-created events.

**Issue Number:**

Fixes #1887 

**Did you add tests for your changes?**
Yes, unit tests have been added to ensure the functionality works as expected.

**Snapshots/Videos:**
Snapshots and a short demonstration video showcasing the feature have been attached for better understanding.
image

**If relevant, did you update the documentation?**
Yes, the relevant sections in the [Talawa-Docs](https://github.com/PalisadoesFoundation/talawa-docs) have been updated to include details about the new feature.
**Summary**

This feature addresses the need for a more user-friendly and comprehensive event calendar.

Problem: The existing calendar lacked the ability to display a list of holidays and events in an organized manner.
Solution:
A list of holidays and events is now displayed below the calendar.
Color-coding has been implemented to distinguish between holidays and events for improved usability.
The UI aligns with the Figma design shared by the team.

**Does this PR introduce a breaking change?**

No, this PR does not introduce any breaking changes.

**Other information**

The Figma design used for this implementation can be found [here](https://www.figma.com/file/dTGr5CnGHvkilU5nwnRXg2/Talawa-Admin-UI-Redesign?type=design&node-id=910-355&mode=design&t=EVDV5SZ2cKnOAc2R-0).
This PR follows the project’s contribution guidelines and is submitted against the develop branch, as recommended.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Release Notes

- **New Features**
	- Enhanced Event Calendar with improved layout for displaying holidays and events.
	- New CSS classes added for better styling and organization of calendar components.

- **Bug Fixes**
	- Removed automatic linting fixes during pre-commit to streamline the commit process.

- **Chores**
	- Updated development dependencies for improved testing and linting processes.
	- Added Vitest type declaration to TypeScript configuration.

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,5 +1,3 @@
-#!/usr/bin/env sh
-. "$(dirname -- "$0")/_/husky.sh"
```
Comment (palisadoes) [inline .husky/pre-commit:1]: Your PR has too many unnecessary files that don’t relate to your issue. You can from the PR by running the commands below from the root directory of the repository

git add -u
git reset HEAD path/to/file
git commit

Please apply these changes to this file.

File: package.json [hunk 2]
```diff
@@ -150,13 +150,13 @@
     "jest": "^27.4.5",
     "jest-localstorage-mock": "^2.4.19",
     "jest-location-mock": "^2.0.0",
-    "jest-preview": "^0.3.1",
+    "jest-preview": "^0.2.3",
```
Comment (palisadoes) [inline package.json:153]: Your PR has too many unnecessary files that don’t relate to your issue. You can from the PR by running the commands below from the root directory of the repository

git add -u
git reset HEAD path/to/file
git commit

Please apply these changes to this file.

File: tsconfig.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "types": ["vite/client", "vite-plugin-svgr/client", "node"],
+    "types": ["vite/client", "vite-plugin-svgr/client", "node", "vitest"],
```
Comment (palisadoes) [inline tsconfig.json:3]: Your PR has too many unnecessary files that don’t relate to your issue. You can from the PR by running the commands below from the root directory of the repository

git add -u
git reset HEAD path/to/file
git commit

Please apply these changes to this file.

**Review by palisadoes — CHANGES_REQUESTED** [review]
Your PR has too many unnecessary files that don’t relate to your issue. You can from the PR by running the commands below from the root directory of the repository

git add -u
git reset HEAD path/to/file
git commit

Please apply these changes to the files indicated.

**Comment by palisadoes** [comment]
Make sure CodeRabbit.ai approves your work.

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. These are the only 2 files that must be updated in your PR.  Please exclude the others using the methodology I mentioned.
2. Also merge your code with the latest upstream/develop-postgres branch. You seem to be reverting recent changes

image

**Comment by gurramkarthiknetha** [comment]
> 1. These are the only 2 files that must be updated in your PR.  Please exclude the others using the methodology I mentioned.
> 2. Also merge your code with the latest upstream/develop-postgres branch. You seem to be reverting recent changes
> 
> image

ok

**Comment by palisadoes** [comment]
1. PTAL. This cannot be reviewed by our reviewers. 
    1. https://github.com/PalisadoesFoundation/talawa-admin/pull/2534#issuecomment-2509622160
1. Please fix the excess number of files submitted too

**Comment by palisadoes** [comment]
1. This has to be closed. 
2. You now only have the unnecessary files in the PR.
3. Please try again when you are ready

**Pull request closed without merging**